water_neuron/src/state.rs

```diff
@@ -541,6 +552,62 @@ impl State {
             .is_none());
     }
 
+    pub fn record_neuron_merge(&mut self, neuron_id: NeuronId) {
+        let withdrawal_id: &u64 = self.neuron_id_to_withdrawal_id.get(&neuron_id).unwrap();
+        assert!(
+            self.withdrawal_to_start_dissolving.remove(withdrawal_id)
+                || (self.withdrawal_to_disburse.remove(withdrawal_id)
+                    && self.to_disburse.remove(&neuron_id).is_some())
+        );
+
+        let withdrawal_request = self
+            .withdrawal_id_to_request
+            .get(withdrawal_id)
+            .unwrap()
+            .clone();
+
+        self.withdrawal_cancelled.insert(*withdrawal_id);
+        assert!(self.neuron_id_to_withdrawal_id.remove(&neuron_id).is_some());
+
+        // Merging the neurons costs two times the ICP ledger transaction fee.
+        // Once to calculate the effects of merging two neurons (step 1).
+        // Once to operate the transaction of the source neuron stake to the target neuron (step 5).
+        // Here is the link to the according merge_neurons function used:
+        // https://github.com/dfinity/ic/blob/714c85c6a4245fb5b39e76f5c8003e6d90e49c4d/rs/nns/governance/src/governance.rs#L2780
+        let icp_stake_e8s = withdrawal_request
+            .icp_due
+            .checked_sub(ICP::from_e8s(2 * DEFAULT_LEDGER_FEE))
+            .expect("ICP due should be greater than 10.");
+        let nicp_stake_value_e8s = self.convert_icp_to_nicp(icp_stake_e8s);
+
+        // 0.5% fee when a withdrawal is cancelled.
+        let nicp_fee = nICP::from_e8s(nicp_stake_value_e8s.0.checked_div(200).unwrap());
```

Comment:
Can't you directly do `nicp_stake_value_e8s.checked_div`?

Reply:
bazel build:
```
pub struct AmountOf<Unit>(pub u64, PhantomData<Unit>);
    | ------------------------- method `checked_div` not found for this struct
```

Plus even if I could, I probably should use
```
nicp_stake_value_e8s.checked_div(nICP::from_e8s(200)).unwrap()

Reply:
So I don't know if it changes a lot

Reply:
The current approach is fine.